packages/react/src/QuestionnaireBuilder/QuestionnaireBuilder.tsx

```diff
@@ -417,6 +420,79 @@ function AnswerBuilder(props: AnswerBuilderProps): JSX.Element {
   );
 }
 
+interface ReferenceTypeProps {
+  item: QuestionnaireItem;
+  onChange: (newOptions: QuestionnaireItemAnswerOption[]) => void;
+}
+
+function ReferenceProfiles(props: ReferenceTypeProps): JSX.Element {
+  const references = props.item.extension ?? [];
+  const referenceProfiles =
+    references.filter((e) => e.url === 'http://hl7.org/fhir/R4/extension-questionnaire-referenceresource.html') ?? [];
+
+  return (
+    <>
+      {referenceProfiles.map((reference: Extension) => {
+        return (
+          <div key={reference.id}>
+            <div
+              style={{
+                display: 'flex',
+                flexDirection: 'row',
+                justifyContent: 'space-between',
+                alignItems: 'center',
+                width: '80%',
+              }}
+            >
+              <div>
+                <TextInput
```

Comment:
Consider codeInput, like we do for the resource navigator in the NavBar (https://github.com/medplum/medplum/blob/4fac613da816dc560f911fcd1fc97c532ec04d97/packages/react/src/AppShell/Navbar.tsx#L106-L120)